tests/test_mc_comm.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+import logging
+from multiprocessing import Queue
+
+from mantarray_desktop_app import create_data_packet
+from mantarray_desktop_app import mc_comm
+from mantarray_desktop_app import McCommunicationProcess
+from mantarray_desktop_app import SERIAL_COMM_MAGIC_WORD_BYTES
+from mantarray_desktop_app import SERIAL_COMM_MAIN_MODULE_ID
+from mantarray_desktop_app import SERIAL_COMM_STATUS_BEACON_PACKET_TYPE
+from mantarray_desktop_app import SERIAL_COMM_STATUS_BEACON_PERIOD_SECONDS
+from mantarray_desktop_app import SerialCommPacketRegistrationTimoutError
+import pytest
+from stdlib_utils import InfiniteProcess
+from stdlib_utils import invoke_process_run_and_check_errors
+
+from .fixtures_mc_comm import fixture_four_board_mc_comm_process
+from .fixtures_mc_simulator import fixture_mantarray_mc_simulator
+from .fixtures_mc_simulator import fixture_mantarray_mc_simulator_no_beacon
+from .helpers import confirm_queue_is_eventually_empty
+from .helpers import confirm_queue_is_eventually_of_size
+from .helpers import put_object_into_queue_and_raise_error_if_eventually_still_empty
+
+__fixtures__ = [
+    fixture_four_board_mc_comm_process,
+    fixture_mantarray_mc_simulator,
+    fixture_mantarray_mc_simulator_no_beacon,
+]
+
+DEFAULT_SIMULATOR_STATUS_CODE = bytes(4)
+
+
+def test_McCommunicationProcess_super_is_called_during_init(mocker):
+    error_queue = Queue()
+    mocked_init = mocker.patch.object(InfiniteProcess, "__init__")
+    McCommunicationProcess((), error_queue)
+    mocked_init.assert_called_once_with(error_queue, logging_level=logging.INFO)
+
+
+def test_McCommunicationProcess_hard_stop__clears_all_queues_and_returns_lists_of_values(
+    four_board_mc_comm_process,
+):
+    mc_process, board_queues, error_queue = four_board_mc_comm_process
+
+    expected = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [9, 10, 11]]
+    expected_error = "error"
+
+    for i, board in enumerate(board_queues):
+        for j, queue in enumerate(board):
+            item = expected[i][j]
+            queue.put(item)
+    confirm_queue_is_eventually_of_size(board_queues[3][2], 1)
+    error_queue.put(expected_error)
+    confirm_queue_is_eventually_of_size(error_queue, 1)
+
+    actual = mc_process.hard_stop()
+    assert actual["fatal_error_reporter"] == [expected_error]
+
+    confirm_queue_is_eventually_empty(board_queues[1][0])
+    confirm_queue_is_eventually_empty(board_queues[2][0])
+    confirm_queue_is_eventually_empty(board_queues[3][0])
+    confirm_queue_is_eventually_empty(board_queues[0][0])
+    confirm_queue_is_eventually_empty(board_queues[0][2])
+
+    assert actual["board_1"]["main_to_instrument_comm"] == [expected[1][0]]
+    assert actual["board_2"]["main_to_instrument_comm"] == [expected[2][0]]
+    assert actual["board_3"]["main_to_instrument_comm"] == [expected[3][0]]
```

Comment:
why were only these 6 assertions chosen? is there some significance, or arbitrary? a comment might be useful here